lib/bindings/c/src/lib.rs

```diff
@@ -306,33 +315,593 @@ pub extern "C" fn dynamo_kv_event_publish_removed(
     }
 }
 
-// Need to setup etcd and nats to run these tests
-// #[cfg(test)]
-// mod tests {
-//     use super::*;
-//     use std::ffi::CString;
-
-//     #[test]
-//     fn test_dynamo_llm_init() {
-//         // Create C-compatible strings
-//         let namespace = CString::new("test_namespace").unwrap();
-//         let component = CString::new("test_component").unwrap();
-
-//         // Call the init function
-//         let result = unsafe {
-//             dynamo_llm_init(
-//                 namespace.as_ptr(),
-//                 component.as_ptr(),
-//                 1,  // worker_id
-//                 32, // kv_block_size
-//             )
-//         };
-
-//         assert_eq!(result as u32, DynamoLlmResult::OK as u32);
-
-//         assert!(WK.get().is_some());
-
-//         let shutdown_result = dynamo_llm_shutdown();
-//         assert_eq!(shutdown_result as u32, DynamoLlmResult::OK as u32);
-//     }
+// KV Router configuration structure for C API
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct DynamoKvRouterConfig {
+    pub overlap_score_weight: f64,
+    pub router_temperature: f64,
+    pub use_kv_events: bool,
+    pub router_replica_sync: bool,
+    pub max_num_batched_tokens: u32,
+}
+
+impl Default for DynamoKvRouterConfig {
+    fn default() -> Self {
+        Self {
+            overlap_score_weight: 1.0,
+            router_temperature: 0.0,
+            use_kv_events: true,
+            router_replica_sync: false,
+            max_num_batched_tokens: 8192,
+        }
+    }
+}
+
+// Helper function to convert DynamoKvRouterConfig to KvRouterConfig
+fn convert_to_kv_router_config(config: DynamoKvRouterConfig) -> KvRouterConfig {
+    KvRouterConfig::new(
+        Some(config.overlap_score_weight),
+        Some(config.router_temperature),
+        Some(config.use_kv_events),
+        Some(config.router_replica_sync),
+        Some(config.max_num_batched_tokens),
+        Some(Some(10000)), // router_snapshot_threshold - use default
+        Some(false),       // router_reset_states - use default
+    )
+}
+
+// KV Router per-request override structure for C API
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct DynamoRouterConfigOverride {
+    pub has_overlap_score_weight: bool,
+    pub overlap_score_weight: f64,
+    pub has_router_temperature: bool,
+    pub router_temperature: f64,
```

Comment:
An `f64` must start on an 8-byte boundary, so there will be seven bytes of padding on both of the bools. The struct will take 32 bytes (1 + 7 pad, 8, 1 + 7 pad, 8).

If you re-order the fields to put the two `f64` first, it will take 24 bytes (8, 8 , 1, 1, 6 padding).

A straightforward rule of thumb is order the fields from biggest to smallest. This is only for `repr(C)`. By default the Rust compiler will re-order the fields for us.